app/controllers/projects_controller.rb

```diff
@@ -38,14 +42,51 @@ def destroy
     @project.destroy
   end
 
+  # GET /projects?search=querystring
+  def search
+    begin
+      @projects = Project.get_project_by_querystring(params[:search]).order(created_at: :desc)
+    rescue ActiveRecord::RecordNotFound => e
+      render json: ErrorSerializer.new(e.message, Rack::Utils.status_code(:not_found)).serialized_json, status: :not_found
+    end
+    @pagy = pagy(@projects)
+    render json: ProjectSerializer.new(@projects, pagination_options).serialized_json
+  end
+
+  # GET /category/category_id/projects
+  def category_projects
+    begin
+      @projects = Project.category_projects(params[:category_id]).order(created_at: :desc)
+    rescue ActiveRecord::RecordNotFound => e
+      render json: ErrorSerializer.new(e.message, Rack::Utils.status_code(:not_found)).serialized_json, status: :not_found
+    end
+    @pagy = pagy(@projects)
+    render json: ProjectSerializer.new(@projects, pagination_options).serialized_json
+  end
+
+  # GET /user/user_id/projects
+  def user_projects
+    begin
+      @projects = Project.user_projects(params[:user_id]).order(created_at: :desc)
```

Comment:
Per listare i progetti per utente:
```Project.joins(:members).where(users: {id: 7})```

Reply:
Effettivamente dal nome del metodo sembra che vengano pescati i progetti relativi ad un utente, ma se vai a vedere cosa fa il metodo vedi che tira fuori solo quelli di cui l'utente è admin, per questo proponevo il cambio di ```user_projects``` in ```created_by```.

In ogni caso anche qua dobbiamo decidere, tra due possibilità:
1) Sono sul profilo di tizio, vedo i progetti creati e i progetti a cui partecipa in due sezioni diverse.
2) Sono sul profilo di tizio e vedo tutti i progetti in lista, quelli creati da lui hanno un badge "Creator", mentre gli altri no. In questo caso restituiamo tutti i progetti e nel frontend implementiamo la logica che metto o meno il badge.

La 2 molto carina, ma ovviamente dobbiamo dare anche la possibilità di filtrare tra le due cose, quindi i due endpoint separati servono comunque.

Reply:
Putroppo non funge, nel caso in cui manchi il parametro la query restituisce nil. This, is so, SAD!

Reply:
Quindi abbiamo deciso che:
**Model**: implementa due metodi
1) ```of_user(user)``` che restituisce tutti i progetti a cui partecipa uno user, indipendentemente dal suo ruolo
2) ```of_user_by_role(user, admin)``` che restituisce tutti i progetti a cui partecipa uno user,  in base al suo ruolo (admin/non-admin)

**Controller**: fa il controllo sulla presenza del parametro :admin e chiama il metodo adatto